@re-/model/demos/generateCode/mapTSfiles.ts

```diff
@@ -1,56 +1,133 @@
-import { fromHere } from "@re-/node"
-import { Project, Statement, ts } from "ts-morph"
+import { Project, SourceFile } from "ts-morph"
+import { Stack } from "../stack.js"
+import { addBlockToMaps, getComment } from "./helpers.js"
 import { regex } from "./regex.js"
 
-const { tagMatch, blockCommentMatch, commentMatch } = regex
+const {
+    blockEndGrab,
+    blockOrStatementGrab,
+    nameGrab,
+    includeGrab,
+    excludeGrab,
+    fileGrab,
+    statementGrab,
+    blockGrab
+} = regex
 
-const getComment = (text: string, range: any) =>
-    text.slice(range.getPos(), range.getEnd())
+const codeDestinations = ["stackBlitz", "tests", "docs"]
 
-const stripCodeTag = (codeTag: string) =>
-    codeTag.replace("@re_place", "").trim()
+const breakdownGrabComment = (comment: string, name?: string) => {
+    const NAME_TAG = "-name:"
+    let optionsDestinations: string[] = []
+    let strippedComment: string
 
-const addFileContentsToMap = (
-    fileText: string,
-    astStatements: Statement<ts.Statement>[],
-    map: Record<string, string>
-) => {
-    for (const content of astStatements) {
-        const range = content.getLeadingCommentRanges()
-        if (range.length) {
-            const foundMatchingComment = getComment(fileText, range[0]).match(
-                tagMatch
+    const commentHasNameLine = comment.match(nameGrab)
+    const nameLine =
+        name ||
+        (!!commentHasNameLine
+            ? commentHasNameLine[0].replace(NAME_TAG, "").trim()
+            : `unnamedGrab${Math.floor(Math.random() * 100000)}`)
```

Comment:
I'd recommend Node's [builtin UUID generator](https://nodejs.org/dist/latest-v18.x/docs/api/crypto.html#cryptorandomuuidoptions) for stuff like this. Can be used like:

```ts
import {randomUUID} from "node:crypto"
const nameLine = `unnamedGrab${randomUUID()}`
```